### PR TITLE
[GEOS-10014] Layers with dimensions are missing on WMS GetCapabilites output when catalog security is set to Challenge

### PR DESCRIPTION
**Pull request by fernandor777**
[GEOS-10014](https://osgeo-org.atlassian.net/browse/GEOS-10014) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geoserver&utm_campaign=badge_info)

This PR introduces a fix and some Junit WMS GetCapabilities integration tests for issue: Layers with dimensions are missing on WMS GetCapabilites output when catalog security is set to Challenge.

Jira ticket:
https://osgeo-org.atlassian.net/browse/GEOS-10014

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or not applicable.**

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the `main` branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [x] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [x] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [x] New unit tests have been added covering the changes
- [x] This PR passes all existing unit tests (test results will be reported by Continuous Integration after opening this PR)
- [x] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by Continuous Integration after opening this PR)
- [x] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates (screenshots, text)
- [x] Commits changing the REST API, or any configuration object, should check if the REST API docs (Swagger YAML files and classic documentation) need to be updated.

### CODE REVIEW

File: src/main/src/main/java/org/geoserver/security/decorators/SecuredFeatureTypeInfo.java [hunk 1]
```diff
@@ -103,14 +107,41 @@ public FeatureType getFeatureType() throws IOException {
...
         }
     }
 
+    /**
+     * Checks if current request is GetCapabilities and returns a new WrapperPolicy with attributes
+     * read allowed to compute the dimensions values. If current request is not a GetCapabilities
+     * one, returns the same policy without changes.
+     */
+    private WrapperPolicy computeWrapperPolicy(Request request) {
+        if (isGetCapabilities(request) && policy.getLimits() instanceof VectorAccessLimits) {
```
Comment (nmco) [inline src/main/src/main/java/org/geoserver/security/decorators/SecuredFeatureTypeInfo.java:124]: I would recommend to add an INFO LOG here explaining what's happening.

File: src/wms/src/test/java/org/geoserver/wms/wms_1_1_1/DimensionsVectorCapabilitiesTest.java [hunk 2]
```diff
@@ -18,13 +24,61 @@
...
+import org.geoserver.security.impl.AbstractUserGroupService;
 import org.geoserver.wms.WMSDimensionsTestSupport;
 import org.junit.Test;
+import org.opengis.filter.Filter;
 import org.w3c.dom.Document;
 import org.w3c.dom.Element;
 
 public class DimensionsVectorCapabilitiesTest extends WMSDimensionsTestSupport {
 
+    /** Add the test resource access manager in the spring context */
```
Comment (nmco) [inline src/wms/src/test/java/org/geoserver/wms/wms_1_1_1/DimensionsVectorCapabilitiesTest.java:42]: End of the sentence missing dot.

**Comment by nmco** [comment]
Just some minor comments, looks fine to me :+1.

File: src/wms/src/test/java/org/geoserver/wms/wms_1_1_1/DimensionsVectorCapabilitiesTest.java [hunk 3]
```diff
@@ -419,4 +473,43 @@ public void testCustomContinuousDate() throws Exception {
...
+        TestResourceAccessManager tam = getResourceAccessManager();
+
+        FeatureTypeInfo featureTypeInfo = catalog.getFeatureTypeByName("sf:TimeElevation");
+        tam.putLimits(
+                "admin2",
+                featureTypeInfo,
+                new VectorAccessLimits(
+                        CatalogMode.CHALLENGE, null, Filter.EXCLUDE, null, Filter.EXCLUDE));
+
+        setupVectorDimension(
```
Comment (aaime) [inline src/wms/src/test/java/org/geoserver/wms/wms_1_1_1/DimensionsVectorCapabilitiesTest.java:489]: Shouldn't this security setup be reverted at the end of the test, or in a @before method?

File: src/main/src/main/java/org/geoserver/security/decorators/SecuredFeatureTypeInfo.java (hunk 1 above)
Comment (fernandor777) [reply src/main/src/main/java/org/geoserver/security/decorators/SecuredFeatureTypeInfo.java:124]: Makes sense, adding it.

File: src/wms/src/test/java/org/geoserver/wms/wms_1_1_1/DimensionsVectorCapabilitiesTest.java (hunk 2 above)
Comment (fernandor777) [reply src/wms/src/test/java/org/geoserver/wms/wms_1_1_1/DimensionsVectorCapabilitiesTest.java:42]: Fixing it.

File: src/wms/src/test/java/org/geoserver/wms/wms_1_1_1/DimensionsVectorCapabilitiesTest.java (hunk 3 above)
Comment (fernandor777) [reply src/wms/src/test/java/org/geoserver/wms/wms_1_1_1/DimensionsVectorCapabilitiesTest.java:489]: Indeed, makes sense, adding on an @After method.

**Pull request merged**